### PR TITLE
fix(openapi): fix sort criteria parameter

### PR DESCRIPTION
**Pull request by RyanHolstien**
* Changes the field in the OpenAPIV3Generator to target the correct string list field
* Deprecates the old sort field
* Adds default behavior for both sortCriteria and sortOrder

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java [hunk 1]
```diff
@@ -570,19 +570,15 @@ private static void addExtraParameters(final Components components) {
         "SortBy" + MODEL_VERSION,
         new Parameter()
             .in(NAME_QUERY)
-            .name("sort")
+            .name("sortCriteria")
```
Comment (david-leifker) [inline metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java:573]: need a note in the release notes section for this

Comment (david-leifker) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java:573]: breaking change

Comment (david-leifker) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java:573]: Yes, the API controller still supports "sort", however client sdks using the spec will break the programmatic apis.

**Review by david-leifker — APPROVED** [review]
pending release note

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java (hunk 1 above)
Comment (RyanHolstien) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/v3/OpenAPIV3Generator.java:573]: It technically doesn't remove the deprecated parameter so if you were previously sending in a single string here _it would not break with these changes_. It only affects the documentation and OpenAPI spec. Will add a note nonetheless.

**Pull request merged**